lib/network.cpp

```diff
@@ -223,7 +227,7 @@ int boinc_socket(int& fd, int protocol) {
     fd = (int)socket(protocol, SOCK_STREAM, 0);
     if (fd < 0) {
         char buf[256];
-        snprintf(buf, sizeof(buf), "%s: socket", time_to_string(dtime()));
+        snprintf(buf, sizeof(buf), "%s: socket failed", time_to_string(dtime()));
```

Comment:
This is done by `perror(buf)` which immediately prints `buf+strerror(errno)` to stderr. It seems the difference between `perror(buf)` and `fprintf(stderr, "%s: %s", buf, strerror(errno))` is that the former is thread-safe and the later not. I would rather stick with the thread-safe variant for now.